scripts/skills/perks/perk_rf_phalanx.nut

```diff
@@ -92,56 +93,24 @@ this.perk_rf_phalanx <- ::inherit("scripts/skills/skill", {
 		return ret;
 	}
 
-	function onGetHitFactors( _skill, _targetTile, _tooltip )
+	function isActorValid( _actor )
 	{
-		if (_skill.isAttack() && !_skill.isRanged() && _targetTile.IsOccupiedByActor && this.getCount() != 0 && ::Reforged.Reach.hasLineOfSight(this.getContainer().getActor(), _targetTile.getEntity()))
-		{
-			_tooltip.push({
-				icon = "ui/tooltips/positive.png",
-				text = this.getName()
-			});
-		}
-	}
-
-	function onGetHitFactorsAsTarget( _skill, _targetTile, _tooltip )
-	{
-		if (_skill.isAttack() && !_skill.isRanged() && this.getCount() != 0 && ::Reforged.Reach.hasLineOfSight(_skill.getContainer().getActor(), this.getContainer().getActor()))
-		{
-			_tooltip.push({
-				icon = "ui/tooltips/negative.png",
-				text = this.getName()
-			});
-		}
-	}
-
-	function onQueryTooltip( _skill, _tooltip )
-	{
-		if (_skill.getID() == "actives.shieldwall" && this.hasAdjacentShieldwall())
+		local actor = this.getContainer().getActor();
+		if (actor.isArmedWithShield())
 		{
-			_tooltip.push({
-				id = 15,
-				type = "text",
-				icon = "ui/icons/special.png",
-				text = ::Reforged.Mod.Tooltips.parseString("Costs " + ::MSU.Text.colorizeMultWithText(this.m.ShieldWallActionPointsMult, {InvertColor = true}) + " [Action Points|Concept.ActionPoints] (to a minimum of " + ::MSU.Text.colorPositive(this.m.ShieldWallActionPointsMininum) + ") and " + ::MSU.Text.colorizeMultWithText(this.m.ShieldWallFatigueMult, {InvertColor = true}) + " [Fatigue|Concept.Fatigue] because of " + ::Reforged.NestedTooltips.getNestedSkillName(this))
-			});
+			return actor.getOffhandItem().getID().find("buckler") == null;
```

Comment:
You don't check for the melee piercing attack in this path. As far as I see, you would enable this perk when wielding your fists with a shield or any weapon really